### PR TITLE
[WIP] Feature/property collection

### PR DESCRIPTION
**Pull request by baumannsven**

### CODE REVIEW

File: src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php [hunk 1]
```diff
@@ -1280,7 +1282,14 @@ protected function parseProperties(ContainerInterface $container)
             if ($definition->hasProperty($propName)) {
                 $property = $definition->getProperty($propName);
             } else {
-                $property = new DefaultProperty($propName);
+                if ((true === array_key_exists('collection', $propInfo['eval']))
+                    ||(true === array_key_exists('columnFields', $propInfo['eval']))
+                    ||(true === array_key_exists('columnsCallback', $propInfo['eval']))
+                ) {
+                    $property = new DefaultCompoundProperty($propName);
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php]: For `collection`, this should add the sub properties here - for the other cases, leave it to MCW.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php [hunk 2]
```diff
@@ -102,6 +103,16 @@ public function removeProperty($property)
      */
     public function hasProperty($name)
     {
+        $chunks = explode('__', $name);
+
+        if ((1 < count($chunks))
+            && (true === $this->hasProperty($chunks[0]))
+        ) {
+            $property = $this->getProperty($chunks[0]);
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: This should trigger a deprecation notice!

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php [hunk 3]
```diff
@@ -112,6 +123,18 @@ public function hasProperty($name)
      */
     public function getProperty($name)
     {
+        $chunks = explode('__', $name);
+
+        if ((1 < count($chunks))
+            && (true === $this->hasProperty($chunks[0]))
+        ) {
+            $property = $this->getProperty($chunks[0]);
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: This should trigger a deprecation notice!

File: src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php [hunk 4]
```diff
@@ -1280,7 +1282,14 @@ protected function parseProperties(ContainerInterface $container)
             if ($definition->hasProperty($propName)) {
                 $property = $definition->getProperty($propName);
             } else {
-                $property = new DefaultProperty($propName);
+                if ((true === array_key_exists('collection', $propInfo['eval']))
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php]: The `true ===` is superfluous here.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php [hunk 5]
```diff
@@ -102,6 +103,16 @@ public function removeProperty($property)
      */
     public function hasProperty($name)
     {
+        $chunks = explode('__', $name);
+
+        if ((1 < count($chunks))
+            && (true === $this->hasProperty($chunks[0]))
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: The `true ===` is superfluous here.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php [hunk 6]
```diff
@@ -112,6 +123,18 @@ public function hasProperty($name)
      */
     public function getProperty($name)
     {
+        $chunks = explode('__', $name);
+
+        if ((1 < count($chunks))
+            && (true === $this->hasProperty($chunks[0]))
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: The `true ===` is superfluous here.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php [hunk 7]
```diff
@@ -112,6 +123,18 @@ public function hasProperty($name)
...
+        ) {
+            $property = $this->getProperty($chunks[0]);
+
+            if (true === $property->hasProperty($name)) {
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: The `true ===` is superfluous here.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/DefaultSubProperty.php [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+ * @license    https://github.com/contao-community-alliance/dc-general/blob/master/LICENSE LGPL-3.0
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\DefaultProperty;
+
+/**
+ * Default implementation of a collection property definition.
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/DefaultSubProperty.php]: phpDoc has to be adapted to the name `subProperty`

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/SubPropertyInterface.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ * @license    https://github.com/contao-community-alliance/dc-general/blob/master/LICENSE LGPL-3.0
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\PropertyInterface;
+
+/**
+ * This interface describes a collection property information.
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/SubPropertyInterface.php]: phpDoc has to be adapted to the name `SubProperty`

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/CompoundPropertyInterface.php [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+ * @license    https://github.com/contao-community-alliance/dc-general/blob/master/LICENSE LGPL-3.0
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection\SubPropertyInterface;
+
+/**
+ * This interface extends the property interface and has information for property with collection.
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/CompoundPropertyInterface.php]: phpDoc has to be adapted to the name `SubProperty`.

Same applies to the doc blocks of all methods.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+ * @filesource
+ */
+
+namespace ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties;
+
+use ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\Collection\SubPropertyInterface;
+use ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException;
+
+/**
+ * Default implementation of a property with collection definition.
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: phpDoc has to be adapted to the name `SubProperty`.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+ */
+class DefaultCompoundProperty extends DefaultProperty implements CompoundPropertyInterface
+{
+    /**
+     * The collection of internal collection property in this property.
+     *
+     * @var array|SubPropertyInterface[]
+     */
+    private $collection = array();
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: Should be named `$properties` and use short array notation `[]`

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+     * {@inheritDoc}
+     */
+    public function clearProperties()
+    {
+        $this->collection = array();
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: Should use short array notation []

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+        return $this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function removeProperty(SubPropertyInterface $property)
+    {
+        $this->collection = array_diff_key($this->collection, array_flip(array($property->getName())));
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: Can be simply `unset($this->collection[$property->getName()]);`.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setProperties(array $properties)
+    {
+        $this->clearProperties();
+
+        foreach ($properties as $collectionProperty) {
```
Comment (discordier) [inline src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: Should be named `$property`

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+    }
+
+    /**
+     * Get a default collection property.
+     *
+     * @param string $name The property name.
+     *
+     * @return DefaultSubProperty
+     */
+    private function getDefaultCollectionProperty($name)
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Has to be adapted to the name `SubProperty`.

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 17]
```diff
@@ -0,0 +1,283 @@
...
+    {
+        return new DefaultSubProperty($name);
+    }
+
+    /**
+     * Get a default propert with collectiony.
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: phpDoc has to be adapted to the name `SubProperty`.

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 18]
```diff
@@ -0,0 +1,283 @@
...
+     * @covers ::getProperties
+     * @covers ::hasProperty
+     */
+    public function testAddDefaultProperty()
+    {
+        $definition = $this->getDefintiion();
+        $property   = $this->getDefaultProperty('dummy-property');
+
+        $this->assertInternalType('array', $definition->getProperties());
+        $this->assertFalse($definition->hasProperty($property->getName()));
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: These lines can simply be:
`$this->assertSame([], $definition->getProperties())`

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 19]
```diff
@@ -0,0 +1,283 @@
...
+        $this->assertFalse($definition->hasProperty('dummy-property-foo'));
+
+        $definition->addProperty($property);
+
+        $this->assertInternalType('array', $definition->getProperties());
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: These lines can simply be:
`$this->assertSame([$property], $definition->getProperties());`

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 20]
```diff
@@ -0,0 +1,283 @@
...
+        $this->assertTrue($definition->hasProperty($property->getName()));
+        $this->assertFalse($definition->hasProperty('dummy-property-foo'));
+    }
+
+    /**
+     * Test exceptions for method add properties.
+     *
+     * @expectedException \ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Annotation can be removed due to `$this->setExpectedException` below

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 21]
```diff
@@ -0,0 +1,283 @@
...
+            'ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException',
+            'Passed value is not an instance of PropertyInterface.'
+        );
+        $definition->addProperty(array());
+    }
+
+    /**
+     * Test exceptions for method add property.
+     *
+     * @expectedException \ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Annotation can be removed due to $this->setExpectedException below

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 22]
```diff
@@ -0,0 +1,283 @@
...
+     * @covers ::addProperty
+     * @covers ::getProperties
+     * @covers ::hasProperty
+     */
+    public function testRemoveDefaultProperty()
+    {
+        $definition = $this->getDefintiion();
+        $property   = $this->getDefaultProperty('dummy-property');
+
+        $this->assertInternalType('array', $definition->getProperties());
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Can be simplified - see above.

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 23]
```diff
@@ -0,0 +1,283 @@
...
+ * @coversDefaultClass \ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\DefaultPropertiesDefinition
+ */
+class DefaultPropertiesDefinitionTest extends TestCase
+{
+    /**
+     * Get the default properties definition.
+     *
+     * @return DefaultPropertiesDefinition
+     */
+    private function getDefintiion()
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Typo

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php [hunk 24]
```diff
@@ -0,0 +1,283 @@
...
+     *
+     * @expectedException \ContaoCommunityAlliance\DcGeneral\Exception\DcGeneralInvalidArgumentException
+     *
+     * @covers ::addProperty
+     * @covers ::hasProperty
+     *
+     * @covers \ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\DefaultCompoundProperty::hasProperty
+     * @covers \ContaoCommunityAlliance\DcGeneral\DataDefinition\Definition\Properties\DefaultCompoundProperty::getProperty
+     */
+    public function testDefaultPropertyWithCollection()
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: Adjust name.

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/Properties/DefaultCompoundPropertyTest.php [hunk 25]
```diff
@@ -0,0 +1,238 @@
...
+    {
+        $property = $this->getProperty('dummy');
+
+        $this->assertInternalType('array', $property->getProperties());
+
+        $collectionProperty = $this->getCollectionProperty('dummy-collection-property');
+
+        $property->addProperty($collectionProperty);
+
+        $this->assertInternalType('array', $property->getProperties());
```
Comment (discordier) [inline tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/Properties/DefaultCompoundPropertyTest.php]: All the assertions here can be simplified as noted above.

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php (hunk 13 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: No we have php 5.4

File: src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php (hunk 4 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php (hunk 1 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/Contao/Dca/Builder/Legacy/LegacyDcaDataDefinitionBuilder.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php (hunk 5 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php (hunk 6 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php (hunk 7 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/DefaultPropertiesDefinition.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/DefaultSubProperty.php (hunk 8 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/DefaultSubProperty.php]: is removed

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/SubPropertyInterface.php (hunk 9 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/Collection/SubPropertyInterface.php]: is removed

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/CompoundPropertyInterface.php (hunk 10 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/CompoundPropertyInterface.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php (hunk 11 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php (hunk 12 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php (hunk 15 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: change

File: src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php (hunk 14 above)
Comment (baumannsven) [reply src/ContaoCommunityAlliance/DcGeneral/DataDefinition/Definition/Properties/DefaultCompoundProperty.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 23 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 16 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 17 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: removed

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 18 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 19 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 20 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 21 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 22 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php (hunk 24 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/DefaultPropertiesDefinitionTest.php]: change

File: tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/Properties/DefaultCompoundPropertyTest.php (hunk 25 above)
Comment (baumannsven) [reply tests/ContaoCommunityAlliance/DcGeneral/Test/DataDefinition/Definition/Properties/DefaultCompoundPropertyTest.php]: change

**Pull request closed without merging**